packages/dds/CMakeLists.txt

```diff
@@ -48,6 +38,162 @@ set(SRC
 
+########################################################################
+# Testing preparation
+option(SOSS-DDS_BUILD_TESTS "Build unitary tests." OFF)
+
+if (SOSS-DDS_BUILD_TESTS)
+    include(CTest)
+    include(../dds-test/thirdparty/catch2/cmake/Catch.cmake)
+
+    add_executable(${PROJECT_NAME}_test_conversion
+        test/main.cpp
+        test/unitary/conversion.cpp
+        src/Conversion.cpp
+    )
+
+    set_target_properties(${PROJECT_NAME}_test_conversion
+        PROPERTIES
+            CXX_STANDARD 14
+            POSITION_INDEPENDENT_CODE ON
+    )
+
+    target_compile_options(${PROJECT_NAME}_test_conversion
+        PRIVATE
+            $<$<CXX_COMPILER_ID:GNU>:-Werror -Wall -Wextra -Wpedantic>
+    )
+
+    target_include_directories(${PROJECT_NAME}_test_conversion
+        PRIVATE
+            "../dds-test/thirdparty/catch2/include"
+            "src"
+    )
+
+    configure_file(
+        test/unitary/resources/soss_types.idl
+        ${PROJECT_BINARY_DIR}/soss_types.idl
+        COPYONLY
+        )
+
+    catch_discover_tests(${PROJECT_NAME}_test_conversion)
+endif()
+
+########################################################################
+# Find FastRTPS
+find_package(fastrtps QUIET)
+
+if(fastrtps_DIR)
+    message(STATUS "Found a fastrtps")
+    target_link_libraries(${PROJECT_NAME}
+        PRIVATE
+            soss::core
+            fastrtps
+            $<$<PLATFORM_ID:Windows>:yaml-cpp>
+    )
+    if (SOSS-DDS_BUILD_TESTS)
+        target_link_libraries(${PROJECT_NAME}_test_conversion
+            PRIVATE
+                fastrtps
+                soss::core
+        )
+    endif()
+
+else()
+    message(STATUS "Using the fastrtps as external project")
+    # Fast-RTPS needs foonathan too
+    unset(foonathan_memory_DIR CACHE) # Reviewer! Do we need to clear CACHE? Copied from uXRCE-DDS-Agent
+    find_package(foonathan_memory QUIET)
+    if(NOT foonathan_memory_FOUND)
+        ExternalProject_Add(foonathan_memory
+            GIT_REPOSITORY
+                https://github.com/foonathan/memory.git
+            GIT_TAG
+                master # ${_foonathan_memory_tag} (set(_foonathan_memory_tag master))
+            PREFIX
+                ${PROJECT_BINARY_DIR}/foonathan_memory
+            INSTALL_DIR
+                ${PROJECT_BINARY_DIR}/temp_install/foonathan_memory
+            CMAKE_CACHE_ARGS
+                -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+                -DFOONATHAN_MEMORY_BUILD_EXAMPLES:BOOL=OFF
+                -DFOONATHAN_MEMORY_BUILD_TESTS:BOOL=OFF
+                -DFOONATHAN_MEMORY_BUILD_TOOLS:BOOL=ON
+                -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
+                -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+            )
+    endif()
+
+    # Fast RTPS.
+    unset(fastrtps_DIR CACHE) # Reviewer! Do we need to clear CACHE? Copied from uXRCE-DDS-Agent
+    find_package(fastrtps QUIET)
+    #find_package(fastrtps ${_fastrtps_version} EXACT QUIET)
```

Comment:
Is this a TODO?

Reply:
Yes, if we want to allow specify a version for these dependencies.

Reply:
ok, could you add a TODO message?

Reply:
```suggestion
    #find_package(fastrtps ${_fastrtps_version} EXACT QUIET) #TODO
```